src/node/heartbeat/sync_blocks.rs

```diff
@@ -248,3 +253,105 @@ pub async fn sync_blocks<B: Blockchain>(
 
     Ok(())
 }
+
+/// returns `true` if the parent hash verification is ok, false if it is not
+fn validate_parent_hash(
+    idx: usize,
+    parent_headers: &[Header],
+    header: &crate::core::Header,
+) -> bool {
+    if idx == 0 || idx - 1 > parent_headers.len() {
```

Comment:
when idx is 0, should the header from `ctx.blockchain.get_header(head.number -1)` be used for parent hash validation?